### DISCUSSION TITLE
[IntelliSense] Not working in Laravel + Vite when config file is module

### DISCUSSION THREAD

**Discussion by czernika** [tailwindlabs/tailwindcss#10503] [Help]
**What version of Tailwind CSS IntelliSense are you using?**

v0.9.7

**What version of Tailwind CSS are you using?**

v3.2.4

**What package manager are you using?**

npm

**What operating system are you using?**

Linux Fedora 37 Workstation (However the app running using Laravel Sail's Docker file)

**Tailwind config**

```js
export default {
    content: [
        './resources/views/**/*.blade.php',
        './resources/js/**/*{.js,.ts,.vue}',
    ],
}
```

**PostCSS config**

```js
import tailwind from 'tailwindcss'
import tailwindCSSNesting from 'tailwindcss/nesting'
import autoprefixer from 'autoprefixer'
import postcssImport from 'postcss-import'

import tailwindConfig from './tailwind.config.js'

export default {
    plugins: [
        postcssImport,
        tailwindCSSNesting,
        tailwind(tailwindConfig),
        autoprefixer,
    ],
}
```

**Vite config**

```js
import { defineConfig } from 'vite'
import laravel, { refreshPaths } from 'laravel-vite-plugin'

import postcss from './postcss.config.js'

export default defineConfig({
    css: {
        postcss,
    },

    plugins: [
        laravel({
            input: ['resources/css/app.css', 'resources/js/app.js'],
            refresh: refreshPaths,
        }),
    ],
})
```

**VS Code settings**

```json
"editor.quickSuggestions": {
    "strings": true
  },
 "files.associations": {
    "*.css": "tailwindcss"
 },
 "tailwindCSS.experimental.classRegex": [
    ["@class\\(([^]*)\\)", "'([^']*)'"]
]
```

Also tried to specify config path via `tailwindCSS.experimental.configFile` with no success 

**Describe your issue**

I need to use value from Tailwind config in my javascript file as it is shown [here](https://tailwindcss.com/docs/configuration#referencing-in-java-script). However using Vite I'm getting a lot of errors and the only solution worked for me is to use configs as modules (basically `.mjs`). All configs are shown beyond

**Steps to reproduce**

1. Install Laravel (Framework version 9.50.2)

```sh
composer create-project laravel/laravel <directory-where-to-install>
```

2. Install Tailwind and change config files

```sh
npm install tailwindcss autoprefixer
npx tailwindcss init -p
```

```js
// tailwind.config.js

export default {
    content: [
        './resources/views/**/*.blade.php',
        './resources/js/**/*{.js,.ts,.vue}',
    ],
}
```

```js
// postcss.config.js

import tailwind from 'tailwindcss'
import autoprefixer from 'autoprefixer'

import tailwindConfig from './tailwind.config.js'

export default {
    plugins: [
        tailwind(tailwindConfig),
        autoprefixer,
    ],
}
```

```js
// vite.config.js

import { defineConfig } from 'vite'
import laravel, { refreshPaths } from 'laravel-vite-plugin'

import postcss from './postcss.config.js'

export default defineConfig({
    css: {
        postcss,
    },

    plugins: [
        laravel({
            input: ['resources/css/app.css', 'resources/js/app.js'],
            refresh: refreshPaths,
        }),
    ],
})
```

```css
// resources/css/app.css

@tailwind base;
@tailwind components;
@tailwind utilities;
```

```js
// resources/js/app.js

import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from '../../tailwind.config.js'

const fullConfig = resolveConfig(tailwindConfig)

console.log(fullConfig.theme.width[4])
```

3. In a `welcome.blade.php` add any utility

![изображение](https://user-images.githubusercontent.com/86893348/216760067-43d131ec-8438-4f5c-9a34-9aa406f0c9a4.png)

I cannot see suggestions, hover and little green square next to color utility. Both `npm run dev` and `npm run build` runs with no error - CSS successfully compiles

If I'll reverse everything back to use `module.exports` - everything works Intellisense included but I cannot see value in a console from app JS file - just errors about mixing commonJS and modules

**Comment by bradlc** [comment]
Hey @czernika. The extension doesn't support ESM config files (see https://github.com/tailwindlabs/tailwindcss-intellisense/issues/565, https://github.com/tailwindlabs/tailwindcss-intellisense/issues/592). Have you tried using CommonJS syntax and renaming the file to `tailwind.config.cjs`?

**Reply by czernika** [reply]
Hi @bradlc!  I tried. However I faced a new issue (not related to Tailwind Intellisense but Vite) - either `Uncaught SyntaxError: ambiguous indirect export: default` (when using `import` - this one discussed [here](https://github.com/tailwindlabs/tailwindcss/discussions/3646) although accepted answer "broke" my build script but [this comment](https://github.com/tailwindlabs/tailwindcss/discussions/3646#discussioncomment-1742991) helped) or `Uncaught ReferenceError: require is not defined` (when using `require()` - it was mentioned [here](https://github.com/vitejs/vite/issues/1241#issuecomment-762367047))

So with ESM configs I can get values from Tailwind config in my JS file using Vite but  that leads to Intellisense to be broken) So I have to face one issue or another I guess

Sorry for duplicate I've missed issue you mentioned when done searching
